### PR TITLE
 Separate Maestro dependencies and manual dependencies into two xml files

### PR DESCRIPTION
**Pull request by kythant**
Unfortunately, Maestro is not happy with how we format our manual dependencies in version.details.xml. 
We need to leave version.details.xml alone and move our manual dependencies into a version.dependencies.xml.

### CODE REVIEW

File: DevCheck.ps1 [hunk 1]
```diff
@@ -801,10 +818,10 @@ function CheckAndSync-Dependencies
         [HashTable]$in
     )
 
-    $automagic = $in["Automagic"]
-    $manual = $in["Manual"]
+    $dependencies = $in["Dependencies"]
```
Comment (DrusTheAxe) [inline DevCheck.ps1:821]: Reversed order

**Review by DrusTheAxe — CHANGES_REQUESTED** [review]

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -35,18 +37,5 @@
     </Dependency>
   </ProductDependencies>
   <ToolsetDependencies>
```
[the remark this answers is not in the document]
Comment (kythant) [reply eng/Version.Details.xml:39]: Thanks for the catch

File: DevCheck.ps1 [hunk 3]
```diff
@@ -650,40 +650,57 @@ function Start-TAEFService
 function Get-DependencyVersions
 {
     # Dependencies are defined in Version.Details.xml
-    #   - <ProductDependencies> = Automagically updated by Maestro
-    #   - <ToolsetDependencies> = Manually updated by human beings
+    #   Automagically updated by Maestro
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1:653]: Not sure what you mean

File: DevCheck.ps1 [hunk 4]
```diff
@@ -650,40 +650,57 @@ function Start-TAEFService
...
         $version = $dependency.Version
         $versions.Add($name, $version)
     }
-    $dependencies.Manual = $versions
+    $dependencies.Dependencies = $versions
 
     if ($Verbose -eq $true)
     {
-        ForEach ($list in @($dependencies.AutoMagic, $dependencies.Manual))
+        ForEach ($list in @($dependencies.Dependencies, $dependencies.Details))
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1]: I've decided to revert the names of these

File: DevCheck.ps1 (hunk 1 above)
Comment (kythant) [reply DevCheck.ps1]: I've decided to revert all of these

File: DevCheck.ps1 [hunk 5]
```diff
@@ -743,8 +760,8 @@ function Build-Dependencies
         [HashTable]$in
     )
 
-    $automagic = $in["Automagic"]
-    $manual = $in["Manual"]
+    $dependencies = $in["Dependencies"]
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1]: I've decided to revert all of these

File: DevCheck.ps1 [hunk 6]
```diff
@@ -851,17 +868,17 @@ function Test-Dependencies
         $global:issues++
         $fatal_errors++
     }
-    $automagic = $dependencies.Automagic
-    $manual = $dependencies.Manual
-    if ([string]::IsNullOrEmpty($automagic))
+    $dependencies = $dependencies.Dependencies
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1]: I've decided to revert all of these

File: DevCheck.ps1 [hunk 7]
```diff
@@ -650,40 +650,57 @@ function Start-TAEFService
...
+    $dependencies.Details = $versions
 
-    # Parse the manual dependencies
+    # Parse The VersionDependencies dependencies
+    $versionDependenciesFileName = Join-Path $path 'Version.Dependencies.xml'
+    Write-Host "Reading $($versionDependenciesFileName)..."
+    $xml = [xml](Get-Content $versionDependenciesFileName -EA:Stop)
+    # Parse the VersionDependencies dependencies
     $versions = [ordered]@{}
+    ForEach ($dependency in $xml.SelectNodes("/Dependencies/ProductDependencies/Dependency"))
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1]: Fixed

File: DevCheck.ps1 [hunk 8]
```diff
@@ -650,40 +650,57 @@ function Start-TAEFService
 function Get-DependencyVersions
 {
     # Dependencies are defined in Version.Details.xml
-    #   - <ProductDependencies> = Automagically updated by Maestro
-    #   - <ToolsetDependencies> = Manually updated by human beings
+    #   Automagically updated by Maestro
+    # Dependencies are defined in Version.Dependencies.xml
+    #   Manually updated by human beings
```
[the remark this answers is not in the document]
Comment (kythant) [reply DevCheck.ps1:655]: ok got what you mean

File: DevCheck.ps1 [hunk 9]
```diff
@@ -946,14 +940,14 @@ Write-Verbose("Processor...$cpu")
...
 
-if (($CheckAll -ne $false) -Or ($CheckVisualStudio -ne $false))
-{
-    $ok = Test-VisualStudio2022Install
-    if ($ok -eq $true)
-    {
-        $null = Test-VisualStudioComponents
-    }
-}
+# if (($CheckAll -ne $false) -Or ($CheckVisualStudio -ne $false))
```
Comment (DrusTheAxe) [inline DevCheck.ps1:944]: Commented out?

Let me guess - it took a while so you commented it out?

No need.

* `devcheck -CheckDependencies` to just do the check
* `devcheck -CheckDependencies -SyncDependencies`  to check and potentially update files

That's fast to run. My testing was usually

1. `devcheck -CheckDependencies` 
2. `devcheck -CheckDependencies -SyncDependencies`
3. `devcheck -CheckDependencies` 
4. `devcheck -CheckDependencies -SyncDependencies`
5. ...

bouncing back and forth between them and seeing all work well, mixed in with actually making problems (e.g. list a package twice in a .xml) and seeing check spot it, sync fix it then check sees all is well. That sort of thing

File: DevCheck.ps1 [hunk 10]
```diff
@@ -1015,4 +1009,4 @@ else
 {
     $n = $global:issues
     Write-Output "$n issue(s) detected"
-}
+}
```
Comment (DrusTheAxe) [inline DevCheck.ps1:1013]: GitHub diff cranky due to no newline at end of file?

**Review by DrusTheAxe — APPROVED** [review]
LGTM modulo the open comments are addressed

**Review by bpulliam — APPROVED** [review]

**Pull request merged**